contracts/staking/PoSValidatorManager.sol

```diff
@@ -425,29 +501,36 @@ abstract contract PoSValidatorManager is
         // a weight update using a higher nonce (which implicitly includes the delegation's weight update)
         // to be used to complete delisting for an earlier delegation. This is necessary because the P-Chain
         // is only willing to sign the latest weight update.
-        require(
-            $._delegatorStakes[delegationID].endingNonce <= nonce,
-            "PoSValidatorManager: nonce does not match"
-        );
+        require(delegator.endingNonce <= nonce, "PoSValidatorManager: nonce does not match");
 
         // Ensure the delegator is pending removed. Since anybody can call this function once
         // end delegation has been initialized, we need to make sure that this function is only
         // callable after that has been done.
         require(
-            $._delegatorStakes[delegationID].status == DelegatorStatus.PendingRemoved,
+            delegator.status == DelegatorStatus.PendingRemoved,
             "PoSValidatorManager: delegation not pending added"
         );
 
-        // Update the delegator status
-        $._delegatorStakes[delegationID].status = DelegatorStatus.Completed;
```

Comment:
I realized we don't actually need a `DelegatorStatus.Completed` since we can clear the delegator's state when we get to this stage. I can remove it from the enum if others thinks that makes sense